### PR TITLE
feat(useWizard): add hook for managing wizard logic for different use cases

### PR DESCRIPTION
**Pull request by YossiSaadi**
https://monday.monday.com/boards/3532714909/pulses/7525076173

image

### CODE REVIEW

File: packages/core/src/hooks/useWizard/__stories__/useWizard.stories.module.scss [hunk 1]
```diff
@@ -0,0 +1,6 @@
+.stepper {
```
Comment (talkor) [inline packages/core/src/hooks/useWizard/__stories__/useWizard.stories.module.scss:1]: I'd stick to the current terminology

Comment (talkor) [reply packages/core/src/hooks/useWizard/__stories__/useWizard.stories.module.scss:1]: Plus why not Flex?

File: packages/core/src/hooks/useWizard/__stories__/useWizard.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+          <Button onClick={next} disabled={!canGoNext}>
+            Next
+          </Button>
+        </Flex>
+      </>
+    );
+  }
+};
+
+export const WithCompletion = {
```
Comment (talkor) [inline packages/core/src/hooks/useWizard/__stories__/useWizard.stories.tsx:69]: I think it's a bit redundant and can be in the overview story

File: packages/core/src/hooks/useWizard/useWizard.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import { useCallback, useState } from "react";
+import { UseWizardProps, UseWizardReturnValue, WizardDirection } from "./useWizard.types";
+
+function useWizard({ initialStep = 0, stepCount, onStepChange, onComplete }: UseWizardProps): UseWizardReturnValue {
```
Comment (talkor) [inline packages/core/src/hooks/useWizard/useWizard.ts:4]: onComplete => onFinish? We use it in the Steps component

**Review by talkor — APPROVED** [review]
🥇

File: packages/core/src/hooks/useWizard/__stories__/useWizard.stories.module.scss (hunk 1 above)
Comment (YossiSaadi) [reply packages/core/src/hooks/useWizard/__stories__/useWizard.stories.module.scss:1]: @talkor Notice it is being used inside
```
        <Steps
          className={styles.stepper}
```

I cannot use Flex

Comment (YossiSaadi) [reply packages/core/src/hooks/useWizard/__stories__/useWizard.stories.module.scss:1]: what do you mean about the terminology, which one

File: packages/core/src/hooks/useWizard/__stories__/useWizard.stories.tsx (hunk 2 above)
Comment (YossiSaadi) [reply packages/core/src/hooks/useWizard/__stories__/useWizard.stories.tsx:69]: agree

File: packages/core/src/hooks/useWizard/__stories__/useWizard.stories.module.scss (hunk 1 above)
Comment (talkor) [reply packages/core/src/hooks/useWizard/__stories__/useWizard.stories.module.scss:1]: Oh ok nevermind, my bad, ignore

**Comment by YossiSaadi** [comment]
@talkor I've fixed your comments

**Pull request merged**